core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/TokenUtils.scala

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import scala.reflect.runtime.currentMirror
+import scala.reflect.runtime.universe._
+import java.time.Instant
+import org.apache.spark.SparkContext
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import spray.json.DefaultJsonProtocol.{StringJsonFormat, jsonFormat3}
+import java.util.UUID
+import com.microsoft.azure.synapse.ml.logging.common.WebUtils._
+import spray.json.{DeserializationException, RootJsonFormat}
+import spray.json.JsonParser.ParsingException
+
+case class MwcToken (TargetUriHost: String, CapacityObjectId: String, Token: String)
+object TokenUtils {
+  var AADToken: String = ""
+  val MwcWorkloadTypeMl = "ML"
+
+  def getAccessToken: String = {
+    if (checkTokenValid(this.AADToken))
+      this.AADToken
+    else {
+      refreshAccessToken()
+      this.AADToken
+    }
+  }
+
+  def getAccessToken(tokenType: String): String = {
+
+    val objectName = "com.microsoft.azure.trident.tokenlibrary.TokenLibrary"
+    val mirror = currentMirror
+    val module = mirror.staticModule(objectName)
+    val obj = mirror.reflectModule(module).instance
+    val objType = mirror.reflect(obj).symbol.toType
+    val methodName = "getAccessToken"
+    val methodSymbols = objType.decl(TermName(methodName)).asTerm.alternatives
+    val argType = typeOf[String]
+
+    val selectedMethodSymbol = methodSymbols.find { m =>
+      m.asMethod.paramLists match {
+        case List(List(param)) => param.typeSignature =:= argType
+        case _ => false
+      }
+    }.getOrElse(throw new NoSuchMethodException(s"Method $methodName with argument type $argType not found"))
+
+    val methodMirror = mirror.reflect(obj).reflectMethod(selectedMethodSymbol.asMethod)
+    methodMirror(tokenType).asInstanceOf[String]
+  }
+
+  def checkTokenValid(token: String): Boolean = {
```

Comment:
instead of checking, returning a boolean, and then throwing based on this boolean. Change this function to something like assertTokenIsValiud and just throw the appropriate errors. Also your errors you are catching and throwing just seem to be  errors from a different part of the code so maybe you can further reduce the code here

Reply:
I am not throwing exceptions based on the boolean. The error might come from FabricTokenParser passed an invalid or no token. So, it's a defensive block of code